tests/app/test_phone_stats_home.py

```diff
@@ -0,0 +1,99 @@
+import synapse
+from synapse.rest.client.v1 import login, room
+
+from tests import unittest
+from tests.unittest import HomeserverTestCase
+
+DAY = 86400
+
+
+class PhoneHomeTestCase(HomeserverTestCase):
+    servlets = [
+        synapse.rest.admin.register_servlets_for_client_rest_resource,
+        room.register_servlets,
+        login.register_servlets,
+    ]
+
+    # Override the retention time for the user_ips table because otherwise it
+    # gets pruned too aggressively for our R30 test.
+    @unittest.override_config({"user_ips_max_age": "365d"})
```

Comment:
Does this mean that R30 doesn't work out of the box?

Reply:
The default is 28d, so out of the box, it does seem like it's slightly broken, yes...

Out of the box, it can only count users who had activity in the last **28** days, where that activity occurred at least 30 days after registration.

The intention would be to count users who had activity in the last **30** days, where that activity occurred at least 30 days after registration, so it's not a MASSIVE difference.

Reply:
I'm just wary of testing a feature by first changing the default config? Can we update the test to work with the default, or do we need to update the default?

Reply:
In my opinion, the default should be changed. (However, I'm not sure how worthwhile that is since the idea is to add R30v2 and eventually remove R30 [I expect]...).
Updating the test to work with the default wouldn't really show what R30 actually means, it'd instead be R28 or something ;p.
I was hoping to basically just add a test to characterise the current behaviour and wasn't intending to change anything, though...

Suppose I could make a copy of the test that uses the default settings? What do you think is preferable?

Reply:
Yeah, I guess its fair to merge as is.

Reply:
Actually, yes, I think we should have a test for the default, since this is for phone home stats.